Firestore/core/src/nanopb/message.h

```diff
@@ -187,6 +187,49 @@ class Message {
   T proto_{};
 };
 
+template <typename T>
+class OptionalMessage {
+ public:
+  OptionalMessage() : message_{absl::nullopt} {
+  }
+
+  OptionalMessage(T message) : message_{message} {};
+
+  OptionalMessage(const OptionalMessage&) = delete;
+  OptionalMessage& operator=(const OptionalMessage&) = delete;
+
+  OptionalMessage(OptionalMessage&& other) noexcept
+      : message_{other.message_ ? std::move(*other.message_) : absl::nullopt} {
+    other.owns_proto_ = false;
+  }
+
+  OptionalMessage& operator=(OptionalMessage&& other) noexcept {
+    if (message_) {
```

Comment:
Why is it impossible to overwrite an empty optional message? Seems like strange semantics.